src/analysis/convex/cone.lean

```diff
@@ -532,3 +533,40 @@ begin
     simp only [convex_cone.mem_mk, mem_set_of_eq, subtype.coe_mk, prod.fst_add, prod.snd_add,
       zero_add, sub_add_cancel] }
 end
+
+/-!
+### The dual cone
+-/
+section dual
+
+variables {H : Type*} [inner_product_space ℝ H] (s t : set H)
+open_locale real_inner_product_space
+
+/-- The dual cone is the cone consisting of all points `y` such that for
+all points `x` in a given set `0 ≤ ⟪ x, y ⟫`. -/
+noncomputable def set.to_dual_cone (s : set H) : convex_cone H := {
```

Comment:
I'm not sure that `set.to_dual_cone` is the best name for this. It's quite a generic name, and in the discussion we already saw that in other contexts other definitions for dual cone will be needed.
But I don't have a better suggestion atm, so let's leave it like this for now.

Reply:
Good point. I've found that there is actually a name for this kind of dual cone using the inner product:
https://en.wikipedia.org/wiki/Dual_cone_and_polar_cone#In_a_Hilbert_space_(internal_dual_cone)